relay.go

```diff
@@ -457,12 +463,13 @@ func (r *Relayer) addRelayItem(isOriginator bool, id, remapID uint32, destinatio
 	if isOriginator {
 		items = r.outbound
 	}
-	item.Timer = time.AfterFunc(ttl, func() { r.timeoutRelayItem(isOriginator, items, id) })
+	item.timeout = r.timeouts.Get()
 	items.Add(id, item)
+	item.timeout.Start(ttl, items, id, isOriginator)
```

Comment:
Wow, this is a pretty subtle race - nice catch!

Reply:
yeah, I was looking at why we were seeing a bunch of spurious logs that led me to this. Unfortunately, I didn't fix it early enough, and the race ended up happening in production:
https://github.com/uber/tchannel-go/issues/687

There's a few other reasons for spurious logs,
https://github.com/uber/tchannel-go/issues/688

Lots of race conditions related to timeouts in the relay paths, so I'm going to try and clean them all up.